Plugins/MatrixPlugin/MatrixLogic.cs

```diff
@@ -0,0 +1,334 @@
+                {
+                    CurrentPerson.ModFirewall = value.ToString();
+                    OnPropertyChanged();
+                    OnPropertyChanged("TotalFirewall");
+                }
+            }
+        }
+
+        public int TotalFirewall => Parse(CurrentPerson.Firewall) + Parse(CurrentPerson.ModFirewall) + (CurrentPerson.Overclocked == "Firewall" ? 1 : 0);
+        
+
+        public int ActionDicePool => GetTotalAttribute(CurrentAction.ActionAttribute) +
+                GetTotalSkill(CurrentAction.ActionSkill) +
+                CurrentAction.ActionModifier +
+                ActionModifier;
+        
+
+        public int DefenceDicePool => GetTotalSkill(CurrentAction.DefenceSkill) +
+                GetTotalAttribute(CurrentAction.DefenceAttribute) +
+                CurrentAction.DefenceModifier +
+                ActionModifier;
+        
+        public int ActionModifier
+        {
+            get => actionModifier;
+            set
+            {
+                actionModifier = value;
+                OnPropertyChanged();
+                OnPropertyChanged("ActionDicePool");
+                OnPropertyChanged("DefenceDicePool");
+            }
+        }
+
+        public int GetTotalMatrixAttribute(string attribute)
+        {
+            switch (attribute)
+            {
+                case "Attack":
+                    return this.TotalAttack;
+                case "Sleaze":
+                    return this.TotalSleaze;
+                case "Data Processing":
+                    return this.TotalDataProcessing;
+                case "Firewall":
+                    return this.TotalFirewall;
+            }
+            return 0;
+        }
+
+        public int GetTotalSkill(string skill)
+        {
+            if (_character.SkillsSection.GetActiveSkill(skill) != null)
+                return _character.SkillsSection.GetActiveSkill(skill).TotalBaseRating;
+            return 0;
+        }
+        public int GetTotalAttribute(string attribute)
+        {
+            if (_character.GetAttribute(attribute) != null)
+                return _character.GetAttribute(attribute).TotalValue;
+            return 0;
+        }
+
+        private int Parse(string value)
+        {
+            int.TryParse(value, out int result);
+            if (result == 0)
+            {
+                CharacterAttrib attribute = _character.GetAttribute(value.Replace("{", "").Replace("}", ""));
+                if (attribute != null)
+                    result = attribute.Base;
+            }
+            return result;
+        }
+
+        public MatrixLogic(Character character, List<MatrixAction> matrixActions)
+        {
+            this._character = character;
+            Actions = matrixActions;
+            Persons = new List<IHasMatrixAttributes>();
+            Software = new List<Gear>();
+            //Load all CyberDecks,Commlinks and Programs to the Lists
```

Comment:
How well does this logic respond to changes in a Character's gear list? I don't see any methods hooking into a character's `Gear::OnCollectionChanged` event handler...

Reply:
Add hooks to a collection changed (https://github.com/chummer5a/chummer5a/pull/4563/commits/cd67c543cb0c14c2a3d4068113bf15cca275a6f9)